app/src/main/java/com/x8bit/bitwarden/ui/tools/feature/generator/GeneratorViewModel.kt

```diff
@@ -267,16 +274,28 @@ class GeneratorViewModel @Inject constructor(
 
     //region Generation Handlers
 
-    private fun loadOptions() {
+    private fun loadOptions(shouldUseStorageOptions: Boolean = false) {
         when (val selectedType = state.selectedType) {
-            is Passcode -> loadPasscodeOptions(
-                selectedType = selectedType,
-            )
+            is Passcode -> {
+                val savedOptionsType: Passcode.PasscodeType? = generatorRepository
+                    .getPasscodeGenerationOptions()
+                    ?.passcodeType
+                    .takeIf { shouldUseStorageOptions }
+                val mainType = savedOptionsType?.let { Passcode(it) } ?: selectedType
+                loadPasscodeOptions(selectedType = mainType)
```

Comment:
Would this be cleaner?
```kotlin
val mainType = if (shouldUseStorageOptions) {
    generatorRepository
        .getPasscodeGenerationOptions()
        ?.passcodeType
        ?.let { Passcode(it) }
        ?: selectedType
} else {
    selectedType
}
loadPasscodeOptions(selectedType = mainType)
```

I am mainly trying to avoid generating the option of we don't need too